### ISSUE TITLE
behaviour of makecpt

### ISSUE THREAD

**Issue by liming-he** [GenericMappingTools/GMT.jl#1198]
See below the output. 

I expect to see different colours for Row #1 and R#2; however, Row #1 and Row #2 are in the same colour.

The first example is from: https://docs.juliahub.com/GMT/EoU0j/0.29.0/makecpt/#Discrete-versus-Continuous-CPT

Is this a bug, or did I misunderstand the example? Many thanks.  -LH.

julia> makecpt(cmap=(:red,:green,:blue), range=[0,100,300,1000], no_bg=true)
Extract of a GMTcpt exposed as a GMTdataset for display. Colors converted to [0-255]
Model: rgb
Color depth: 24
3×9 GMTdataset{Float64, 2}
 Row │      r1       g1       b1       r2       g2       b2    alpha       z1       z2 
     │ Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64 
─────┼─────────────────────────────────────────────────────────────────────────────────
   1 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0      0.0    100.0
   2 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0    100.0    300.0
   3 │     0.0    255.0      0.0      0.0    255.0      0.0      0.0    300.0   1000.0

julia> topo = makecpt(cmap=(:red, :blue, :green, :yellow), range= [0.0, 0.01, 0.05, 0.1, 0.2]) #make colorbar  other options: color=:polar
Extract of a GMTcpt exposed as a GMTdataset for display. Colors converted to [0-255]
Model: rgb
Color depth: 24
4×9 GMTdataset{Float64, 2}
 Row │      r1       g1       b1       r2       g2       b2    alpha       z1       z2 
     │ Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64 
─────┼─────────────────────────────────────────────────────────────────────────────────
   1 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0     0.0      0.01
   2 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0     0.01     0.05
   3 │     0.0      0.0    255.0      0.0      0.0    255.0      0.0     0.05     0.1
   4 │   255.0    255.0      0.0    255.0    255.0      0.0      0.0     0.1      0.2

**Comment by joa-quim** [comment]
Had a very quick look (need to go out) and yes, there seems to be some error. Try
```
C = makecpt(cmap=(:red,:green,:blue), range="0,100,300,1000", no_bg=true)
imshow(C)
```

Note that the GMT.jl location is not that anymore. Type ``@? makecpt`` to see the new docs.

(I'll come back when I find what the first form is not working well)

**Comment by liming-he** [comment]
Thanks for the quick response.  It is something relating to the non-linear scale.

**Comment by joa-quim** [comment]
Indeed, the behavior depends on the numbers, which is not right. It really looks a bug upstream (GMT binary)
```
# good

julia> makecpt(cmap=(:red,:green,:blue), range=[0 100 200 450])
Extract of a GMTcpt exposed as a GMTdataset for display. Colors converted to [0-255]
Model: rgb
Color depth: 24
3×9 GMTdataset{Float64, 2}
 Row │      r1       g1       b1       r2       g2       b2    alpha       z1       z2
     │ Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64
─────┼─────────────────────────────────────────────────────────────────────────────────
   1 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0      0.0    100.0
   2 │     0.0    255.0      0.0      0.0    255.0      0.0      0.0    100.0    200.0
   3 │     0.0      0.0    255.0      0.0      0.0    255.0      0.0    200.0    450.0

# broken

julia> makecpt(cmap=(:red,:green,:blue), range=[0 100 200 451])
Extract of a GMTcpt exposed as a GMTdataset for display. Colors converted to [0-255]
Model: rgb
Color depth: 24
3×9 GMTdataset{Float64, 2}
 Row │      r1       g1       b1       r2       g2       b2    alpha       z1       z2
     │ Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64
─────┼─────────────────────────────────────────────────────────────────────────────────
   1 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0      0.0    100.0
   2 │   255.0      0.0      0.0    255.0      0.0      0.0      0.0    100.0    200.0
   3 │     0.0      0.0    255.0      0.0      0.0    255.0      0.0    200.0    451.0
```

**Comment by liming-he** [comment]
Thanks!

**Comment by liming-he** [comment]
Thanks for tracking the error.

**Comment by joa-quim** [comment]
Fixed in #1200

**Comment by liming-he** [comment]
thanks!
